### PR TITLE
github: Download Solaris10 box Image If Missing

### LINKED ISSUE #2694: "Vagrant Playbook Checker / Solaris" github check failing for all PRs
Sample check link: https://github.com/adoptium/infrastructure/runs/7750861669?check_suite_focus=true
```
Bringing machine 'adoptopenjdkSol10' up with 'virtualbox' provider...
==> adoptopenjdkSol10: Box 'solaris10' could not be found. Attempting to find and install...
    adoptopenjdkSol10: Box Provider: virtualbox
    adoptopenjdkSol10: Box Version: >= 0
==> adoptopenjdkSol10: Box file was not detected as metadata. Adding it directly...
==> adoptopenjdkSol10: Adding box 'solaris10' (v0) for provider: virtualbox
    adoptopenjdkSol10: Downloading: solaris10

An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

Couldn't open file /Users/runner/work/infrastructure/infrastructure/ansible/solaris10
```

### PR DESCRIPTION
**Pull request by steelhead31**
Download the adoptium solaris10 box image if not present in vagrant.

Fixes #2694 

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] VPC/QPC not applicable for this PR

### CODE REVIEW

**Comment by zdtsw** [comment]
maybe try rebase my [PR](https://github.com/adoptium/infrastructure/pull/2733#) so you can test it?

File: .github/workflows/build_vagrant.yml [hunk 1]
```diff
@@ -21,6 +21,19 @@ jobs:
...
         brew install --cask virtualbox
 
+    - name: Install Solaris 10 Box If Not Present
+      run: |
+        if [ `vagrant box list|grep ^solaris10|wc -l` -gt 0 ]
+        then
+          echo "Box Exists - Do Nothing"
+        else
+          echo "No Box - Download From Jenkins And Import"
+          wget https://ci.adoptopenjdk.net/userContent/vagrant/Solaris10.box.gz
```
Comment (zdtsw) [inline .github/workflows/build_vagrant.yml:31]: should this be verified? like checksum

Comment (Haroon-Khel) [reply .github/workflows/build_vagrant.yml]: +1 on checksum verification

Comment (steelhead31) [reply .github/workflows/build_vagrant.yml]: Checksum validation added.

**Review by zdtsw — APPROVED** [review]

**Review by Haroon-Khel — APPROVED** [review]
LGTM

**Pull request merged**